jiant/scripts/postproc/xtreme/README.md

```diff
@@ -0,0 +1,369 @@
+# XTREME Running / Submission Guide
+
+This guide will walk through the full process for evaluating XLM-R on all XTREME tasks. It consists of the following steps:
+
+* [Download Model](#download-model)
+* [Download Data](#download-data)
+* [Tokenize and Cache Data](#tokenize-and-cache-data)
+* [Generate Run Configs](#generate-run-configs)
+* [Train/Run Models](#trainrun-models)
+
+You can also choose to just run one of the tasks, instead of the whole benchmark.
+
+Before we begin, be sure to set the following environment variables:
+```bash
+BASE_PATH=/home/zp489/scratch/working/v1/2008/15_xtreme
+MODEL_TYPE=xlm-roberta-large
+``` 
+
+## Download Model
+
+First, we download the XLM-R model.
+
+```bash
+# Download XLM-R Large
+python jiant/proj/main/export_model.py \
+    --model_type ${MODEL_TYPE} \
+    --output_base_path ${BASE_PATH}/models/${MODEL_TYPE}
+```
+
+## Download Data
+
+Next, we download the XTREME data. We also need to download the MNLI and SQuAD datasets as training data for XNLI, XQuAD and MLQA.
+
+```bash
+python jiant/scripts/download_data/runscript.py \
+    download \
+    --benchmark XTREME \
+    --output_path ${BASE_PATH}/tasks/
+python jiant/scripts/download_data/runscript.py \
+    download \
+    --tasks mnli squad_v1 \
+    --output_path ${BASE_PATH}/tasks/
+```
+
+## Tokenize and Cache Data
+
+Now, we preprocess our data into a tokenized cache. We need to do this across all languages for each XTREME task. Somewhat tediously (and this will come up again), different tasks have slightly different phases (train/val/test) available, so we have slightly different configurations for each.
+
+### XNLI (uses MNLI for training)
+```bash
+for LG in ar bg de el en es fr hi ru sw th tr ur vi zh; do
```

Comment:
Any cleaner ways to handle the different languages? Can we store this in a constants file? Will a user always know which languages they want to run/what is supported by XTREME?

Reply:
My issue is that it's going to be ugly regardless, and I would prefer everything be explicit rather than trying the abstract and handle the different cases.

E.g.

1. For some tasks, there is a separate training set. For others, it's the English version of the XTREME task.
2. Some tasks don't have separate tests sets.
3. Some tasks have different languages available in val and test.

I realize it is currently verbose but it is very clear to the reader what is happening, and how to quickly find what they're looking for, and I would recommended keeping it this way. This also makes it easier for a user who wants to just run a subset of tasks.